src/Swashbuckle.AspNetCore.SwaggerGen/XmlComments/XmlCommentsRequestBodyFilter.cs

```diff
@@ -13,97 +13,165 @@ public class XmlCommentsRequestBodyFilter : IRequestBodyFilter
         public XmlCommentsRequestBodyFilter(XPathDocument xmlDoc) : this(XmlCommentsDocumentHelper.CreateMemberDictionary(xmlDoc))
         {
         }
-
         internal XmlCommentsRequestBodyFilter(IReadOnlyDictionary<string, XPathNavigator> xmlDocMembers)
         {
             _xmlDocMembers = xmlDocMembers;
         }
 
         public void Apply(OpenApiRequestBody requestBody, RequestBodyFilterContext context)
         {
-            var parameterDescription =
-                context.BodyParameterDescription ??
-                context.FormParameterDescriptions.FirstOrDefault((p) => p is not null);
+            var bodyParameterDescription = context.BodyParameterDescription;
 
-            if (parameterDescription is null)
+            if (bodyParameterDescription is not null)
             {
-                return;
+                var propertyInfo = bodyParameterDescription.PropertyInfo();
+                if (propertyInfo is not null)
+                {
+                    ApplyPropertyTagsForBody(requestBody, context, propertyInfo);
+                    return;
+                }
+                var parameterInfo = bodyParameterDescription.ParameterInfo();
+                if (parameterInfo is not null)
+                {
+                    ApplyParamTagsForBody(requestBody, context, parameterInfo);
+                    return;
+                }
             }
-
-            var propertyInfo = parameterDescription.PropertyInfo();
-            if (propertyInfo is not null)
+            else
             {
-                ApplyPropertyTags(requestBody, context, propertyInfo);
-                return;
-            }
+                var numberOfFromForm = context.FormParameterDescriptions?.Count() ?? 0;
+                if (requestBody.Content?.Count is 0 || numberOfFromForm == 0)
+                {
+                    return;
+                }
 
-            var parameterInfo = parameterDescription.ParameterInfo();
-            if (parameterInfo is not null)
-            {
-                ApplyParamTags(requestBody, context, parameterInfo);
+                foreach (var formParameter in context.FormParameterDescriptions)
+                {
+                    if (formParameter.PropertyInfo() is not null || formParameter.Name is null)
+                    {
+                        continue;
+                    }
+
+                    var parameterFromForm = formParameter.ParameterInfo();
+                    if (parameterFromForm is null)
+                    {
+                        continue;
+                    }
+
+                    foreach (var item in requestBody.Content.Values)
+                    {
+                        if ((item?.Schema?.Properties?.TryGetValue(formParameter.Name, out var value) ?? false)
```

Comment:
The level of `?` is a bit excessive - I think it would be more readable to just check for null first:

```csharp
if (item?.Schema?.Properties is not { } properties)
{
    continue;
}
```